### ISSUE TITLE
downcastToAttributeElement's callback is executed with null

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-engine#1419]
```js
		editor.conversion.for( 'editingDowncast' )
			.add( downcastAttributeToElement( { model: 'linkHref', view: ( href, writer ) => {
				const safeHref = isSafeUrl( href || '' ) ? href : '#';

				return createLinkElement( safeHref, writer );
			} } ) );
```

Why did I need `href || ''`? Because `href`'s sometimes `null`.

How can `href` be `null`? 

https://github.com/ckeditor/ckeditor5-engine/blob/797cd97702c77bc03d39d7de0d16b14adc1a2ed6/src/conversion/downcast-converters.js#L757

Because `elementCreator()` is called with the old attr value which is `null` when the attribute is applied. 

The `elementCreator()` callback should not be called when we know that the attribute is applied. I assume that we treat `null` as lack of value so that should not be a problem.

cc @scofalik

**Comment by Reinmar** [comment]
Am I right that we can avoid calling `elementCreator()` in such a case?

**Comment by Reinmar** [comment]
BTW, the same should be true for removing attr – in this case `newAttrValue` will be `null` so we should not call `elementCreator()` with it.

**Comment by pjasiun** [comment]
I agree there is no reason to call creator in such case. I believe if the old model attribute value is `undefined` or `null` we do not need to unwrap anything in the view and if the new model attribute value is `undefined` or `null` we do not need to wrap anything.
